runtime/connectors/connectors.go

```diff
@@ -135,3 +146,41 @@ func (s *Source) PropertiesEquals(o *Source) bool {
 
 	return true
 }
+
+// SplitFileRecursive recursively strips file extension and returns file name and extension
+func SplitFileRecursive(path string) (string, string) {
+	fullExt := filepath.Ext(path)
+	fullName := strings.TrimSuffix(path, fullExt)
+
+	for true {
+		ext := filepath.Ext(fullName)
+		if ext == "" {
+			break
+		}
+		fullExt = ext + fullExt
+		fullName = strings.TrimSuffix(path, fullExt)
+	}
+
+	return fullName, fullExt
+}
+
+func CreateTempAndCopy(name string, ext string, writer io.ReadCloser) (string, error) {
```

Comment:
I would also move this function to the `fileutil` package. Some other nits:

- Maybe call it `CopyToTempFile` instead and take the `io.ReadCloser` as the first param?
- The reader be called `reader` or `r` and not `writer`
- This function shouldn't close the reader – closing should be done by the opener when possible or otherwise clearly documented. So we can also change the type of it to just `io.Reader`